pkg/cluster/kubernetes/detect.go

```diff
@@ -34,12 +34,6 @@ func DetectLowestVersion(ctx context.Context, cluster UpgradeProvider, options U
 	}
 
 	var version *semver.Version
-	if options.ToVersion != "" {
```

Comment:
I'm not sure why this block was here (?), but it fails when _downgrading_ via `talosctl upgrade-k8s`: as we start detection with `--to` version, it's always smaller than any version in the cluster.

E.g. cluster runs v1.23.1, and I want to downgrade to v1.23.0:

```
talosctl upgrade-k8s --to=1.23.0
```

This will always auto-detect running version as 1.23.0, while correct should be 1.23.1

cc @Unix4ever  did I miss anything?

Reply:
Yeah, I think your fix is correct.